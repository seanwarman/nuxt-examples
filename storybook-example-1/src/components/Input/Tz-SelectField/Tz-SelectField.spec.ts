/**
 * @vitest-environment jsdom
 */

import { mount, VueWrapper } from '@vue/test-utils';
import TzSelectField from './Tz-SelectField.vue';
import { expect, describe, it } from 'vitest';
import type { Option } from '../types';

const text = "I'm a button";
let wrapper: VueWrapper;
const name = 'testName';
const options: Option[] = [
  {
    name: 'option 1 ',
    value: 1
  },
  {
    name: 'Another option',
    value: 2
  },
  {
    name: 'The third option is a long one',
    value: 3
  },
  {
    name: 'Last choice',
    value: 4
  }
];

describe('TzSelectField', () => {
  it('renders correctly', () => {
    wrapper = mount(TzSelectField, {
      props: {
        name,
        options,
        slots: {
          default: text
        }
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('sets options', () => {
    wrapper = mount(TzSelectField, {
      props: {
        name,
        options
      },
      slots: {
        default: text
      }
    });
    const selectElement: HTMLSelectElement = wrapper.find('select').element;
    expect(selectElement.options.length).toBe(options.length);
    expect(parseInt(selectElement.options[2].value, 10)).toBe(options[2].value);
    expect(selectElement.options[2].text).toBe(options[2].name);
  });

  it('sets the name attribute', () => {
    wrapper = mount(TzSelectField, {
      props: {
        name,
        options
      },
      slots: {
        default: text
      }
    });
    const selectElement: HTMLSelectElement = wrapper.find('select').element;
    expect(selectElement.name).toBe(name);
  });

  it('sets the disabled attribute', () => {
    wrapper = mount(TzSelectField, {
      props: {
        name,
        options,
        disabled: true
      },
      slots: {
        default: text
      }
    });
    const selectElement: HTMLSelectElement = wrapper.find('select').element;
    expect(selectElement.disabled).toBeTruthy();
  });

  it('sets an initial selected option', () => {
    wrapper = mount(TzSelectField, {
      props: {
        name,
        options,
        value: 1
      },
      slots: {
        default: text
      }
    });
    const select = wrapper.find('select');
    expect(parseInt(select.element.value, 10)).toBe(1);
  });
});
