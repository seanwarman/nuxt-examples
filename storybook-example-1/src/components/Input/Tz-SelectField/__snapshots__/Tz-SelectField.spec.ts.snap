// Vitest Snapshot v1

exports[`TzSelectField > renders correctly 1`] = `
<div
  class="flex flex-col gap-1.5"
  slots="[object Object]"
>
  <label
    class="text-xs font-black uppercase tracking-wider text-gray-800"
    for="testName"
  >
     
    <!--v-if-->
  </label>
  <!--v-if-->
  <!-- NOTE: ive used a simple styled select box for now - just seems simpler -->
  <!-- &lt;tz-select :options="options" :name="name" :disabled="disabled" :default-state="defaultState"&gt;&lt;/tz-select&gt; -->
  <select
    class="block w-full select-none appearance-none rounded-md border-gray-300 py-3 hover:cursor-pointer focus:border-dark focus:ring-1 focus:ring-dark disabled:cursor-auto text-gray-400"
    id="testName"
    name="testName"
  >
    <!--v-if-->
    
    <option
      value="1"
    >
      option 1 
    </option>
    <option
      value="2"
    >
      Another option
    </option>
    <option
      value="3"
    >
      The third option is a long one
    </option>
    <option
      value="4"
    >
      Last choice
    </option>
    
  </select>
  <!--v-if-->
</div>
`;
